Libraries/WNContainers/inc/optional.h

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2017, WNProject Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE.txt file.
+
+#pragma once
+
+#ifndef __WN_CONTAINERS_OPTIONAL_H__
+#define __WN_CONTAINERS_OPTIONAL_H__
+
+#include <optional>
+
+namespace wn {
+namespace containers {
+
+using std::optional;
```

Comment:
Would this be more of a core thing? It's not really a container

Reply:
Yeah we could put it in core. Will do right now